[#12] Add CapacityInputView and CustomSlider

Seon-U opened the pull request:
## 📝 작업 내용
- capacity를 입력받는 화면을 생성했습니다.
- 커스텀 슬라이더를 추가했습니다.
- 직접 입력을 받는 바텀시트 화면과 임시 팁 화면을 생성했습니다.
- 팁 화면 구성을 위한 객체인 RoundedCorner를 추가했습니다.
- 뷰 연결 및 체크를 위해 현재 진행상황을 올립니다.
- 브랜치 이슈로 타 브랜치 작업한 파일을 새 브랜치로 이동했습니다.

> 이번 PR에서 작업한 내용을 간략히 설명해주세요(이미지 첨부 가능)

### 스크린샷 (선택)

## 💬 리뷰 요구사항(선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
> 
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?

Seon-U committed:
feat: [#12] Add CapacityInputView and CustomSlider

- 브랜치 이슈로 타 브랜치 작업한 파일을 새 브랜치로 이동했습니다.
- capacity를 입력받는 화면을 생성했습니다.
- 직접 입력을 받는 바텀시트 화면과 임시 팁 화면을 생성했습니다.
- 팁 화면 구성을 위한 객체인 RoundedCorner를 추가했습니다.
- 커스텀 슬라이더를 추가했습니다.
- 임시값들이 많아 뷰 연결 및 체크를 위해 현재 진행상황을 올립니다.

zaehorang commented on MoonCrystal/MoonCrystal/SelectCapacity/CapacityDirectInputView.swift:
```diff
@@ -0,0 +1,78 @@
+    @State var tempCapacity: Double = 0.0
+    
+    var favoriteIdol = "최애"
+    let title = "를 위해 \n몇 GB 정리할까요?"
```
이 밑으로 한 줄 띄우면 좋을 거 같아요!

Seon-U replied:
수정완료!

zaehorang commented on MoonCrystal/MoonCrystal/SelectCapacity/RoundedCorner.swift:
```diff
@@ -0,0 +1,25 @@
+}
+
+extension View {
+    func cornerRadius(_ radius: CGFloat, corners: UIRectCorner) -> some View {
+        clipShape(RoundedCorner(radius: radius, corners: corners))
+    }
```
위에 구조체 내부에 메서드를 선언하지 않고 view를 extension해서 선언하신 이유가 있나요?

Seon-U replied:
뷰 익스텐션이 조금 더 접근이 편하지 않나 싶었는데 어차피 특정할 때만 사용하는 객체니까, 생각해보고 수정할게요!

zaehorang commented on MoonCrystal/MoonCrystal/SelectCapacity/DummyView.swift:
```diff
@@ -0,0 +1,22 @@
+//
+//  Created by seonu kim on 7/30/24.
+//
+
```
해당 파일 지워주세요!

Seon-U replied:
수정 완료!

zaehorang commented on MoonCrystal/MoonCrystal/SelectCapacity/CapacitySettingView.swift:
```diff
@@ -0,0 +1,142 @@
+        return MediaCapacityConverter.capacityToPhoto(capacity: Int(selectedCapacity) * 1_073_741_824)
+    }
+    var videoTime : Int {
+        return MediaCapacityConverter.capacityToTime(capacity: Int(selectedCapacity) * 1_073_741_824, format: videoFormat)
```
비슷해 보이는 로직인데 해당 모델에 메서드로 선언하는게 어떠신가요?

zaehorang commented on MoonCrystal/MoonCrystal/SelectCapacity/CapacitySettingView.swift:
```diff
@@ -0,0 +1,142 @@
+import SwiftUI
+
+struct CapacitySettingView: View {
+    @State var selectedCapacity : Double = 0
```
타입 선언부 세미클론 붙여주세요

Seon-U replied:
수정완료!

zaehorang commented on MoonCrystal/MoonCrystal/SelectCapacity/CapacitySettingView.swift:
```diff
@@ -0,0 +1,142 @@
+                
+                VStack(alignment: .leading, spacing: 16) {
+                    HStack(alignment: .center, spacing: 0) {
+                        Text("동영상 \(videoTime.minutesToHoursAndMinutes().hours)시간 \(videoTime.minutesToHoursAndMinutes().minutes)분 찍을 수 있어요")
```
피그바 뷰를 보면 시간만 있을 때는 분이 보이지 않습니다! 
분기처리가 필요해 보여요

zaehorang commented on MoonCrystal/MoonCrystal/SelectCapacity/CapacitySettingView.swift:
```diff
@@ -0,0 +1,142 @@
+                .padding(.horizontal, 20)
+                
+                NavigationLink {
+                    //정리 시작 페이지로 이동
```
이건 약간 팁인데 //TODO: ~~ 로 쓰면 보기 더 편해요

zaehorang commented on MoonCrystal/MoonCrystal/SelectCapacity/CustomSlider.swift:
```diff
@@ -0,0 +1,101 @@
+            }
+        }
+    }
+    func paddingForValue(scaleFactor : CGFloat, thumbSize : CGFloat, step : Int) -> Double {
+        return scaleFactor * CGFloat(step) - thumbSize / 2
+    }
+    private func offsetForValue(value: Int, scaleFactor: CGFloat, thumbSize: CGFloat, minValue: CGFloat, maxValue: CGFloat, isLast: Bool) -> CGFloat {
```
줄바꿈 부타드려요!

Seon-U replied:
개행완료!

zaehorang reviewed (commented):
코멘트 확인 부탁드려요!

leesungkug commented on MoonCrystal/MoonCrystal/SelectCapacity/CustomSlider.swift:
```diff
@@ -0,0 +1,101 @@
+    var step : Int = 20
+    var sliderRange: ClosedRange<Double> = 0...80
+    var trackColor = Color.gray100
+    var thumbColor = Color.pink300
```
color값을 변수에 할당해서 사용하신 이유가 있을까요? 하나의 객체에만 적용되는 컬러면 변수에 할당안하고 하시는건 어떠신가요?

Seon-U replied:
나중에 색이 변동될 때를 고려했던 것 같아요. 지금은 디자인이 고정됐으니 그에 맞춰 수정 완료했습니다!

leesungkug commented on MoonCrystal/MoonCrystal/SelectCapacity/CustomSlider.swift:
```diff
@@ -0,0 +1,101 @@
+
+// 일부 함수나 변수의 경우에 추후 간격 조정 및 수정을 위해 삭제하지 않고 내버려두었습니다.
+struct CustomSlider: View {
+    @Binding var value: Double
```
value 명칭을 어떤 value를 뜻하는지 명확하게 하는건 어떠신가요?

Seon-U replied:
selectedCapacity로 통일했어요!

leesungkug commented on MoonCrystal/MoonCrystal/SelectCapacity/CustomSlider.swift:
```diff
@@ -0,0 +1,101 @@
+            let lower = sliderRange.lowerBound
+            let sliderVal = min(((self.value - lower) * scaleFactor + minValue), maxValue)
+            
+            VStack(alignment: .leading, spacing: 0) {
+                ZStack {
+                    RoundedRectangle(cornerRadius: radius)
+                        .frame(height: trackHeight)
+                        .foregroundColor(trackColor)
+//                                            .padding(.horizontal, trackPadding)
+                    HStack {
+                        Circle()
+                            .foregroundColor(thumbColor)
+                            .frame(width: thumbSize, height: thumbSize)
+                            .offset(x: sliderVal - thumbSize / 2)
+                            .shadow(color: .black.opacity(0.12), radius: 6.5, x: 0, y: 6)
+                            .shadow(color: .black.opacity(0.12), radius: 2, x: 0, y: 0.5)
+                            .gesture(
+                                DragGesture(minimumDistance: 0)
+                                    .onChanged { change in
+                                        let translation = change.translation.width
+                                        if (abs(translation) < 0.1) {
+                                            self.lastCoordinateValue = sliderVal
+                                        }
+                                        let nextCoordinateValue = max(minValue, min(maxValue, self.lastCoordinateValue + translation))
+                                        self.value = (nextCoordinateValue - minValue) / scaleFactor + lower
+                                    }
+                            )
+                        Spacer()
+                    }
+                }
```
시간되면 나중에 뷰를 나누면 가독성이 더 좋을 것 같습니다

Seon-U replied:
시간이... 되면 좋겠어요! 일단 힘내보겠습니다! 화이팅! ;)

Seon-U committed:
Feat: [#12] switch ment based on Minute, clean up the code

- 분의 존재에 따라 보여지지 않도록 설정하고, Text굵기 조절함
- DummyView 삭제힘
- CustomSlider의 한번만 사용되는 변수 삭제함
- 변수 사이 간격 조절힘
- 함수 사이 개행힘

zaehorang reviewed (approved):
고생하셨습니다 🐯
